client/src/lib/components/sandbox/play/phase-input/MeleePhaseInput.svelte

```diff
@@ -0,0 +1,119 @@
+<script lang="ts">
+	import { MinaArenaClient } from '$lib/mina-arena-graphql-client/MinaArenaClient';
+
+	export let game: Game;
+	export let currentPlayer: string;
+	export let rerender: () => {};
+
+	const minaArenaClient = new MinaArenaClient();
+
+	const playerPieces = game.gamePieces.filter((p) => {
+		return p.gamePlayer.player.minaPublicKey === currentPlayer;
+	});
+
+  const enemyPieces = game.gamePieces.filter((p) => {
+		return p.gamePlayer.player.minaPublicKey !== currentPlayer;
+	});
+
+  let meleeAttacks: Record<string, MeleeAttackAction> = {};
+
+  const updateMeleeAttackTarget = (e: Event, p: GamePiece) => {
+    const target = e.target as HTMLInputElement;
+    const targetGamePieceId = target.valueAsNumber;
+
+    if (meleeAttacks[p.id]) {
+      meleeAttacks[p.id].action.targetGamePieceId = targetGamePieceId;
+    } else {
+      meleeAttacks[p.id] = {
+        gamePieceId: p.id,
+        action: {
+          targetGamePieceId: targetGamePieceId,
+          diceRoll: {
+            publicKey: { x: 'xValue', y: 'yValue' },
+            cipherText: 'supersecret',
+            signature: { r: 'rValue', s: 'sValue' },
+          }
+        }
+      }
+    }
+  }
+
+	const submitPhase = async () => {
+		await minaArenaClient.submitMeleePhase(
+			currentPlayer,
+			game.id,
+			game.currentPhase!.id,
+			Object.values(meleeAttacks)
+		);
+		await rerender();
+	};
+
+	const calculateDistance = (p: GamePiece) => {
+    const meleeAttack = meleeAttacks[p.id];
+		if (meleeAttack) {
+      const targetGamePieceId = meleeAttack.action.targetGamePieceId;
+      const targetPiece = enemyPieces.find(enemyPiece => enemyPiece.id.toString() === targetGamePieceId.toString());
+
+      if (targetPiece) {
+        return Math.sqrt(
+          (p.coordinates.x - targetPiece.coordinates.x) ** 2 +
+            (p.coordinates.y - targetPiece.coordinates.y) ** 2
+        ).toFixed(2);
+      } else {
+        return 'Not found!';
+      }
+		} else {
+			return 0;
+		}
+	};
+</script>
+
+<br>
+<div>
+	<table>
+		<tr>
+			<th>Your Piece</th>
+			<th>Current Location</th>
+			<th>Max Range</th>
+      <th>Hit</th>
+      <th>Wound</th>
+      <th>AP</th>
+      <th>Damage</th>
+			<th>Target Piece</th>
+			<th>Distance</th>
+		</tr>
+		{#each playerPieces || [] as piece}
+			<tr>
+				<td>{piece.playerUnit.name || 'Bob'} ({piece.playerUnit.unit.name}, ID: {piece.id})</td>
+				<td>{piece.coordinates.x}, {piece.coordinates.y}</td>
+        <td>5</td>
```

Comment:
This should be modified in a future PR to pull the constant from the `mina-arena-server` package. That's blocked behind https://github.com/trumpet-zk-ignite-1/mina-arena-server/pull/31.

Reply:
Another option could be an API endpoint which is given a unit and returns enemy units in melee range